src/scratch/ScratchCostume.as

```diff
@@ -344,9 +346,16 @@ public class ScratchCostume {
 		r.left = Math.floor(r.left);
 		r.height += Math.floor(r.top) - r.top;
 		r.top = Math.floor(r.top);
-		var image:BitmapData = new BitmapData(
-				Math.max(1, Math.ceil(r.width) + 1), Math.max(
-						1, Math.ceil(r.height) + 1), true, 0);
+		var desiredWidth: int = Math.max(0, Math.ceil(r.width));
+		var desiredHeight: int = Math.max(0, Math.ceil(r.height));
+		if (desiredWidth >= DisplayObjectContainerIn3D.texSizeMax
+				|| desiredHeight >= DisplayObjectContainerIn3D.texSizeMax) {
+			var factor:Number = (DisplayObjectContainerIn3D.texSizeMax - 1) / Math.max(desiredWidth, desiredHeight);
+			desiredWidth *= factor;
+			desiredHeight *= factor;
+		}
+		// TODO: figure out why we add 1 to each dimension in addition to using Math.ceil above
+		var image:BitmapData = new BitmapData(desiredWidth + 1, desiredHeight + 1, true, 0);
```

Comment:
Luckily the code below is already set up to use whatever size `image` happens to get here.